source/Geant/track/TrackState.hpp

```diff
@@ -198,9 +200,15 @@ struct TrackState
         // to join the entries with second separator and then join the paired
         // entries with the first separator
         using apply_t = tim::apply<std::tuple<std::string>>;
-        auto&& labels = std::make_tuple("addr", "status", "step", "Pstep", "pos", "dir", "time");
-        auto&& values = std::make_tuple(&t, t.fStatus, t.fStep, t.fPhysicsState.fPstep, t.fPos, t.fDir, t.fTime);
-        os << apply_t::join(", ", "=", labels, values);
+        auto&& labels1 = std::make_tuple("addr", "status", "#steps", "step", "Pstep", "snext", "safety", "time", "totLen");
+        auto&& values1 = std::make_tuple(&t, t.fStatus, t.fHistoryState.fNsteps, t.fStep, t.fPhysicsState.fPstep,
+					 t.fGeometryState.fSnext, t.fGeometryState.fSafety, t.fTime, t.fTotalLength);
+        os << apply_t::join(", ", "=", labels1, values1) <<"\n";
+        auto&& labels2 = std::make_tuple("addr", "pos", "dir", "boundary");
+        auto&& values2 = std::make_tuple(&t, t.fPos, t.fDir, t.fGeometryState.fBoundary);
+        os << apply_t::join(", ", "=", labels2, values2) <<", fPaths:\n";
+	t.fGeometryState.fPath->Print();
+	t.fGeometryState.fNextpath->Print();
```

Comment:
```suggestion
        t.fGeometryState.fPath->Print();
        t.fGeometryState.fNextpath->Print();
```
Make sure there is no 'tab' character in the file to avoid misaligned on other's editor.